### DISCUSSION TITLE
Scripting: How to get an integer value int(x) and append to a string variable

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#16922] [Q&A] [answered]
I would like to get the integer part of a floating variable and append it to a textstring.

eg:
```
myvar=14.50
mysvar=<integeter value of myvar> + "other text"

```
If I do 
`mysvar=int(myvar) + "other string"`
it does not work

**Comment by barbudor** [comment, marked as answer]
`int(myvar)` return a number, not a string. 
You need to use:
```
s(x) = explicit conversion from number x to string may be preceded by precision digits e.g. s(2.2x) = use 2 digits before and after decimal point
```
which would looks like:
```
mysvar=s(int(myvar)) + "other string"
```

**Reply by jkwim** [reply]
I tried:

```
mystr=s(int(tvar))+"test"
print tvar %tvar% mystr %mystr%

```
and get:

`tvar 1.10 mysvar 1.00test`

So it appears that because int(x) is picking the correct part but as a float, the string conversion happens on the float.

I ended up using sb():

```
mysvar=s(int(tvar))
mysvar=sb(mysvar 0 1)
print tvar %tvar% mysvar %mysvar%

```

And I get the desired output.

Thanks

**Reply by barbudor** [reply]
Indeed there is nothing like "integer variable" in Tasmota, neither in Rules nor Scripting
Except for Berry on ESP32 which have integer variables
